classes/checkout/CheckoutAddressesStep.php

```diff
@@ -71,25 +71,6 @@ public function handleRequest(array $requestParams = array())
             }
         }
 
-        if (isset($requestParams['id_address_delivery'])) {
```

Comment:
I don't know all the whole process work, but removing these tests from here may break an existing feature don't you think?

Reply:
It's not remove but move to the bottom. You will able to set a delivery adress even if you're not in confirm context.

Reply:
How is that even possible if you're moving these checks inside a whole `} elseif (isset($requestParams['confirm-addresses'])) {`?

Reply:
Because these variables exist only in `confirm-addresses` context